.github/workflows/container.yaml

```diff
@@ -57,8 +59,48 @@ jobs:
             "type=ref,event=tag"
             "type=ref,event=pr"
 
-      - name: Build and push Docker image
-        uses: docker/build-push-action@v6
+      - name: Build Docker image
+        uses: docker/build-push-action@v5
+        with:
+          context: .
+          file: Dockerfile.${{ matrix.dockerfile }}
+          load: true
+          push: false
+          tags: ${{ env.TEST_TAG }}
+          labels: ${{ steps.meta.outputs.labels }}
+          build-args: |
+            VARIANT=${{ matrix.variant }}
+
+      - name: Checkout fakemachine (for unit tests)
+        uses: actions/checkout@v4
+        with:
+          repository: 'go-debos/fakemachine'
+          path: src/fakemachine
+
+      - name: Run fakemachine unit tests in Docker image
+        run: |
+          docker run --rm \
+                     --workdir /mnt \
+                     --mount "type=bind,source=$(pwd),destination=/mnt" \
+                     ${{ env.TEST_TAG }} \
+                     bash -c "ls -la && cd src/fakemachine && go test -v ./... --backend=qemu"
```

Comment:
is it also worth testing qemu&kvm backends for debos/fakemachine ? OR since that should already be tested maybe better to just prefer kvm IF it's supported on the host for faster CI runs